### PR TITLE
Added SurveyObj class

### PR DESCRIPTION
**Pull request by SteRiccio**

### CODE REVIEW

File: src/survey/surveyObj.ts [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+  uuid: string
+  id?: number
+  props?: P
+
+  constructor(uuid: string) {
+    this.uuid = uuid
+  }
+}
+
+export abstract class NodeDefObj<P extends NodeDefProps = NodeDefProps> extends ArenaObj<P> {
```
Comment (ramirobg94) [inline src/survey/surveyObj.ts:27]: puff, this code smell to the 2000s, Java is back...

File: src/survey/surveyObj.ts [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+  analysis?: boolean
+  dateCreated?: string
+  dateModified?: string
+  deleted?: boolean
+  draft?: boolean
+  meta?: NodeDefMeta
+  propsAdvanced?: NodeDefPropsAdvanced
+  propsAdvancedDraft?: NodeDefPropsAdvanced
+  propsDraft?: P
```
Comment (ramirobg94) [inline src/survey/surveyObj.ts:39]: P ?

Comment (SteRiccio) [reply src/survey/surveyObj.ts:39]: P extends NodeDefProps is the generic for the props

File: src/survey/surveyObj.ts (hunk 1 above)
Comment (SteRiccio) [reply src/survey/surveyObj.ts:27]: yes, but writing nodeDef.parent is much simpler than writing Surveys.getNodeDefParent({ survey, nodeDef }) or Surveys.getNodeDefParent(nodeDef)(survey), isn't it?!

**Pull request closed without merging**